### PR TITLE
Added job_notification_emails and deidentify actions in google_data_loss_prevention_job_trigger

### PR DESCRIPTION
**Pull request by abheda-crest**
Added `job_notification_emails` and `deidentify` actions in the `google_data_loss_prevention_job_trigger` resource.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dlp: added `inspect_job.actions.job_notification_emails` and `inspect_job.actions.deidentify`  fields to `google_data_loss_prevention_job_trigger` resource
```

### CODE REVIEW

File: mmv1/products/dlp/JobTrigger.yaml [hunk 1]
```diff
@@ -452,8 +458,109 @@ properties:
...
+                - !ruby/object:Api::Type::Array
+                  name: 'fileTypesToTransform'
+                  description: |
+                    List of user-specified file type groups to transform. If specified, only the files with these filetypes will be transformed.
+
+                    If empty, all supported files will be transformed. Supported types may be automatically added over time. 
+                    
+                    If a file type is set in this field that isn't supported by the Deidentify action then the job will fail and will not be successfully created/started.
+                  item_type: !ruby/object:Api::Type::Enum
+                    name: 'undefined'
```
Comment (zli82016) [inline mmv1/products/dlp/JobTrigger.yaml:516]: The  name should be fileType?

File: mmv1/products/dlp/JobTrigger.yaml [hunk 2]
```diff
@@ -452,8 +458,109 @@ properties:
...
+              exactly_one_of:
+                - save_findings
+                - pub_sub
+                - publish_findings_to_cloud_data_catalog
+                - publish_summary_to_cscc
+                - job_notification_emails
+                - deidentify
+              allow_empty_object: true
+              send_empty_value: true
+              properties: []
```
Comment (zli82016) [inline mmv1/products/dlp/JobTrigger.yaml:479]: Just want to double check that the value of jobNotificationEmails should be {} ?

File: mmv1/third_party/terraform/tests/resource_data_loss_prevention_job_trigger_test.go [hunk 3]
```diff
@@ -101,6 +101,57 @@ func TestAccDataLossPreventionJobTrigger_dlpJobTriggerPubsub(t *testing.T) {
 	})
 }
 
+func TestAccDataLossPreventionJobTrigger_dlpJobTriggerJobNotificationEmails(t *testing.T) {
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_data_loss_prevention_job_trigger_test.go:104]: Can you please move these two new tests to examples in JobTrigger.yaml file? https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/dlp/JobTrigger.yaml#L28

In this file mmv1/third_party/terraform/tests/resource_data_loss_prevention_job_trigger_test.go, can you please add tests for updating field?

File: mmv1/products/dlp/JobTrigger.yaml (hunk 2 above)
Comment (abheda-crest) [reply mmv1/products/dlp/JobTrigger.yaml:493]: In the [API docs](https://cloud.google.com/dlp/docs/reference/rest/v2/Action#JobNotificationEmails), there is no information about the object of `jobNotificationEmails`. Moreover, I've tried sending `jobNotificationEmails` as {} and it's working fine. I've also checked this with the API explorer and it is not suggesting any fields inside `jobNotificationEmails`.

Following is the payload I've used to configure `jobTriggers` with `jobNotificationEmails` action:
```
---[ REQUEST ]---------------------------------------
{
 "jobTrigger": {
  "description": "Description for the job_trigger created by terraform",
  "displayName": "TerraformDisplayName",
  "inspectJob": {
   "actions": [
    {
     "jobNotificationEmails": {},
     "publishFindingsToCloudDataCatalog": null,
     "publishSummaryToCscc": null
    }
   ],
   "inspectTemplateName": "sample-inspect-template",
   "storageConfig": {
    "cloudStorageOptions": {
     "fileSet": {
      "url": "gs://mybucket/directory/"
     }
    }
   }
  },
  "status": "HEALTHY",
  "triggers": [
   {
    "schedule": {
     "recurrencePeriodDuration": "86400s"
    }
   }
  ]
 }
}

---[ RESPONSE ]--------------------------------------
{
  "name": "projects/{project_name}/jobTriggers/{job_trigger_id}",
  "displayName": "TerraformDisplayName",
  "description": "Description for the job_trigger created by terraform",
  "inspectJob": {
    "storageConfig": {
      "cloudStorageOptions": {
        "fileSet": {
          "url": "gs://mybucket/directory/"
        }
      }
    },
    "inspectTemplateName": "sample-inspect-template",
    "actions": [
      {
        "jobNotificationEmails": {}
      }
    ]
  },
  "triggers": [
    {
      "schedule": {
        "recurrencePeriodDuration": "86400s"
      }
    }
  ],
  "createTime": "2023-04-13T08:22:50.200795Z",
  "updateTime": "2023-04-13T08:22:50.200795Z",
  "status": "HEALTHY"
}
```

File: mmv1/third_party/terraform/tests/resource_data_loss_prevention_job_trigger_test.go (hunk 3 above)
Comment (abheda-crest) [reply mmv1/third_party/terraform/tests/resource_data_loss_prevention_job_trigger_test.go]: I've moved these tests to examples and also added the update tests.

**Comment by zli82016** [comment]
> Hey @zli82016, I've worked on a PR to further extend same resource and have Sam Levenick as reviewer and it seems like he's not available. Could you please review that PR? Here's the link: #7677

Sure. I will review the PR #7677  later.

File: mmv1/products/dlp/JobTrigger.yaml (hunk 1 above)
Comment (zli82016) [reply mmv1/products/dlp/JobTrigger.yaml]: You can click the link in Diff report to check the generated resource in the provider. 

https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7687-old..auto-pr-7687

`file_type` is not there.

File: mmv1/products/dlp/JobTrigger.yaml (hunk 2 above)
Comment (zli82016) [reply mmv1/products/dlp/JobTrigger.yaml:493]: I cannot find the doc for this field, either. I asked the service team and am waiting for their response. Sorry for holding the PR a little bit.

Comment (zli82016) [reply mmv1/products/dlp/JobTrigger.yaml:493]: I checked with the service team and they confirmed that this field should be an empty json object.

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**